templates/CRM/Contact/Form/Relationship.tpl

```diff
@@ -144,10 +144,113 @@
       CRM.$(function($) {
         var
           $form = $("form.{/literal}{$form.formClass}{literal}"),
-          relationshipData = {/literal}{$relationshipData|@json_encode}{literal};
-        $('[name=relationship_type_id]', $form).change(function() {
+          $relationshipTypeSelect = $('[name=relationship_type_id]', $form),
+          relationshipData = {},
+          contactTypes = {};
+
+        $('body').on('crmOptionsEdited', 'a.crm-option-edit-link', refreshRelationshipData);
```

Comment:
Create a `init` function, it would help to organise the code, also place other initialisation code inside.
```
(function init () {
  $('body').on('crmOptionsEdited', 'a.crm-option-edit-link', refreshRelationshipData);
  refreshRelationshipData().done(...)
  $relationshipTypeSelect.change(function() {
})()
```